fsharp-backend/tests/testfiles/string.tests

```diff
@@ -385,6 +385,50 @@ String.split_v0 "x22y" "2" = ["x"; ""; "y"]
 String.split_v0 "x1y" "1" = ["x"; "y"]
 String.split_v0 "6a6aa6aaa6aaaa" "a" = ["6"; "6"; ""; "6"; ""; ""; "6"; ""; ""; ""]
 String.split_v0 "👨‍❤️‍💋‍👨👩‍👩‍👧‍👦🏳️‍⚧️‍️🇵🇷" "" = ["👨‍❤️‍💋‍👨"; "👩‍👩‍👧‍👦"; "🏳️‍⚧️‍️"; "🇵🇷"]
+String.split_v0 "👨‍👩‍👧‍👦" "👩" = ["👨‍"; "‍👧‍👦"]
+
+String.split_v1 "hello😄world" "😄" = [ "hello"; "world" ]
+String.split_v1 "hello&&&&world" "&&&&" = [ "hello"; "world" ]
+String.split_v1 "hello34564world34564sun" "😄" = [ "hello34564world34564sun" ]
+String.split_v1 "hello34564world34564sun" "34564" = [ "hello"; "world"; "sun" ]
+String.split_v1 "" "34564" = [""]
+String.split_v1 "34564" "" = [ "3"; "4"; "5"; "6"; "4" ]
+String.split_v1 "🧑🏽‍🦰🧑🏼‍💻🧑🏻‍🍼✋✋🏻✋🏿" "🧑🏻‍🍼" = [ "🧑🏽‍🦰🧑🏼‍💻"; "✋✋🏻✋🏿" ]
+String.split_v1 "Z̤͔ͧ̑̓ä͖̭̈̇lͮ̒ͫǧ̗͚̚o̙̔ͮ̇͐̇" "Z̤͔ͧ̑̓ä͖̭̈̇lͮ̒ͫǧ̗͚̚o̙̔ͮ̇͐̇" = [""; ""]
+String.split_v1 "﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽﷽" "﷽﷽﷽﷽" = [ ""; ""; ""; ""; "" ]
+String.split_v1 "👱👱🏻👱🏼👱🏽👱🏾👱🏿" "👱🏼👱🏽" = [ "👱👱🏻"; "👱🏾👱🏿" ]
+String.split_v1 "🧟‍♀️🧟‍♂️" "👱🏽" = [ "🧟‍♀️🧟‍♂️" ]
+String.split_v1 "👨‍❤️‍💋‍👨👩‍👩‍👧‍👦🏳️‍⚧️‍️🇵🇷" "👩‍👩‍👧‍👦"= [ "👨‍❤️‍💋‍👨"; "🏳️‍⚧️‍️🇵🇷" ]
+String.split_v1 "żółw🧑🏽‍🦰🧑🏻‍🍼✋✋🏻✋🏿" "🧑🏽‍🦰" = [ "żółw"; "🧑🏻‍🍼✋✋🏻✋🏿" ]
+String.split_v1 "Z̤͔ͧ̑̓ä͖̭̈̇lͮ̒ͫǧ̗͚̚o̙̔ͮ̇͐̇" "Z̤͔ͧ̑̓ä͖̭̈̇lͮ̒ͫ" = [ "", "ǧ̗͚̚o̙̔ͮ̇͐̇" ]
+String.split_v1 "666666" "6" = [""; ""; ""; ""; ""; ""; ""]
+String.split_v1 "55555" "5" = [""; ""; ""; ""; ""; ""]
+String.split_v1 "4444" "4" = [""; ""; ""; ""; ""]
+String.split_v1 "333" "3" = [""; ""; ""; ""]
+String.split_v1 "22" "2" = [""; ""; ""]
+String.split_v1 "1" "1" = [""; ""]
+String.split_v1 "" "" = []
+String.split_v1 "666666x" "6" = [""; ""; ""; ""; ""; ""; "x"]
+String.split_v1 "55555x" "5" = [""; ""; ""; ""; ""; "x"]
+String.split_v1 "4444x" "4" = [""; ""; ""; ""; "x"]
+String.split_v1 "333x" "3" = [""; ""; ""; "x"]
+String.split_v1 "22x" "2" = [""; ""; "x"]
+String.split_v1 "1x" "1" = [""; "x"]
+String.split_v1 "x666666" "6" = ["x"; ""; ""; ""; ""; ""; ""]
+String.split_v1 "x55555" "5" = ["x"; ""; ""; ""; ""; ""]
+String.split_v1 "x4444" "4" = ["x"; ""; ""; ""; ""]
+String.split_v1 "x333" "3" = ["x"; ""; ""; ""]
+String.split_v1 "x22" "2" = ["x"; ""; ""]
+String.split_v1 "x1" "1" = ["x"; ""]
+String.split_v1 "x666666y" "6" = ["x"; ""; ""; ""; ""; ""; "y"]
+String.split_v1 "x55555y" "5" = ["x"; ""; ""; ""; ""; "y"]
+String.split_v1 "x4444y" "4" = ["x"; ""; ""; ""; "y"]
+String.split_v1 "x333y" "3" = ["x"; ""; ""; "y"]
+String.split_v1 "x22y" "2" = ["x"; ""; "y"]
+String.split_v1 "x1y" "1" = ["x"; "y"]
+String.split_v1 "6a6aa6aaa6aaaa" "a" = ["6"; "6"; ""; "6"; ""; ""; "6"; ""; ""; ""; ""]
+String.split_v1 "👨‍❤️‍💋‍👨👩‍👩‍👧‍👦🏳️‍⚧️‍️🇵🇷" "" = ["👨‍❤️‍💋‍👨"; "👩‍👩‍👧‍👦"; "🏳️‍⚧️‍️"; "🇵🇷"]
+String.split_v1 "👨‍👩‍👧‍👦" "👩" = ["👨‍"; "‍👧‍👦"]
```

Comment:
I don't think it should do this.

Reply:
> Check out #2972 along with this

Good point. I think our investigation here into what other langs do, and the various edge cases, means I'm happy with what we chose to do here and to ignore #2972.

Reply:
To be clearer, we shouldn't split below the "character" (egc) size.